### PR TITLE
Milestone: Friendship

### PR DESCRIPTION
**Pull request by cyonii**
- Modelled friendship concept
- Tested all Models

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -1,7 +1,8 @@
 class Comment < ApplicationRecord
-  validates :content, presence: true, length: { maximum: 200,
-                                                too_long: '200 characters in comment is the maximum allowed.' }
+  validates :content,
+            presence: true,
+            length: { maximum: 200, too_long: '200 characters in comment is the maximum allowed.' }
```
Comment (ghost) [inline app/models/comment.rb]: The validation test for this is not complete. You have omitted tests for length. Hint: You can use [shoulda Matchers](http://matchers.shoulda.io/docs/v4.0.1/Shoulda/Matchers/ActiveModel.html) for this.
Make a similar change for `post.rb` and `user.rb` files.

File: app/models/user.rb [hunk 2]
```diff
@@ -1,12 +1,27 @@
...
+    friends_array.compact
+  end
+
+  def outgoing_friend_requests
+    friendships.map { |f| f if f.pending? }.compact
+  end
+
+  def incoming_friend_requests
+    inverse_friendships.map { |f| f if f.pending? }.compact
+  end
```
Comment (ghost) [inline app/models/user.rb]: Add tests for the above methods. If however, you can make them private then you don't have to write tests for them.

File: app/controllers/friendships_controller.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  private
+
+  def friendship_params
+    params.require(:friendship).permit(:friend_id)
+  end
+
+  def existing_friendship?
+    # current_user.friendships.include?()
+  end
```
Comment (ghost) [inline app/controllers/friendships_controller.rb]: Remove the commented code.

**Review by ghost — CHANGES_REQUESTED** [review]
# Status: Changes Required 🛑 

## Hello team 👋

**Great job on this milestone 💪. However, there are a few changes needed before the project can be approved.**

### Note: I will use examples to explain.
1). **I have two users, Jen and Mercy. Jen sends Mercy a friend request. Mercy can not be able to accept the request because she only sees the cancel request button.**
2). **Please add a profile page for every user. A user profile should already be there from the forked repo. Ensure that when a user receives a friend request, they can see it on their profile page as well. See [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png) from pathwright.**
3). **A user should not see another user's posts unless they are confirmed friends. See [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png) from pathwright.**

4). **I get the error below when I run `rubocop -a`.**
rubocop

5). **Add tests for your views.**

6). **Remove logic in `layouts/application.html.erb` file in views directory.**

7). **Fix the linter errors.**

8). **Check for more in the comments below.**

## Happy coding 🥂.

**PS: You can reach me on slack (Rose Wanjohi).**

**Comment by cyonii** [comment]
About Number 2: Clicking on a user's name will take you to their profile. Kindly let me know if this is not expected case.

**Comment by mimipeshy** [comment]
### STATUS : APPROVED :partying_face: 

Hey team @cyonii :wave: :wave: , great job on working on all requirements needed. This project is therefore approved. Go ahead and merge :smile: 
All the best in the next milestones!!!

Regards
Peris Ndanu

**Pull request merged**